### DISCUSSION TITLE
Pcm displays abnormal data, when used in combination with memguard.

### DISCUSSION THREAD

**Discussion by K1ngRay** [opcm/pcm#570] [Q&A]
Memgurad : https://github.com/heechul/memguard.git

When I insmod memguard, the IPC becomes -1, and all other indicators are abnormal. As long as I rmmod memguard, the PCM is normal again.

Where may this be the problem? How should I correct it?

![image](https://github.com/intel/pcm/assets/33110686/b6123e5a-a314-4bc3-8926-544d546db90a)
![image](https://github.com/intel/pcm/assets/33110686/ee7d1473-96a2-4753-9e08-272352829abf)

**Comment by rdementi** [comment]
could you try to run PCM without RDT (https://github.com/intel/pcm/blob/master/doc/ENVVAR_README.md):

```
export PCM_NO_RDT=1
pcm
```

does it help or you still see the contention issue?

**Comment by K1ngRay** [comment]
> could you try to run PCM without RDT (https://github.com/intel/pcm/blob/master/doc/ENVVAR_README.md):
> 
> ```
> export PCM_NO_RDT=1
> pcm
> ```
> 
> does it help or you still see the contention issue?

don't work! IPC still is -1.00.
![image](https://github.com/intel/pcm/assets/33110686/b2bfabba-a66b-4e12-88d4-53af770f1e9b)
![image](https://github.com/intel/pcm/assets/33110686/4e4f31da-7cfe-4e3d-aa5f-f0417cf08124)
![image](https://github.com/intel/pcm/assets/33110686/4371f8fc-2898-40c6-839a-3266c70f102f)

**Comment by rdementi** [comment]
I had a look at the source code of memguard. Same as PCM, memguard relies on exclusive access to the performance monitoring unit (CPU performance counters):
https://github.com/heechul/memguard/blob/bcb08a4ad64316e193d635b535984795a9e9cc50/memguard.c#L755

ALso memguard seems to have a higher priority such that it revoked access to CPU performance counters from pcm.

**Comment by K1ngRay** [comment]
> I had a look at the source code of memguard. Same as PCM, memguard relies on exclusive access to the performance monitoring unit (CPU performance counters): https://github.com/heechul/memguard/blob/bcb08a4ad64316e193d635b535984795a9e9cc50/memguard.c#L755
> 
> ALso memguard seems to have a higher priority such that it revoked access to CPU performance counters from pcm.

Oh, I see. Thank you very much!
